pkg/webhook/merge_on_review_approval_and_status_checks.go

```diff
@@ -40,27 +40,41 @@ func (h *autoMerger) HandleEvent(w http.ResponseWriter, payload interface{}, ghC
 		return h.handlePullRequestEvent(w, event, ghClientFunc)
 	case *github.StatusEvent:
 		return h.handleStatusEvent(w, event, ghClientFunc)
+	case *github.PullRequestReviewEvent:
+		return h.handlePullRequestReviewEvent(w, event, ghClientFunc)
 	default:
 		return nil
 	}
 }
 
+func (h *autoMerger) handlePullRequestReviewEvent(w http.ResponseWriter, event *github.PullRequestReviewEvent, ghClientFunc GitHubIntegrationsClientFunc) error {
+	if strings.ToLower(event.Review.GetState()) != approvedReviewState {
+		return nil
+	}
+
+	return h.mergePRFromPullRequestEvent(event.Installation.GetID(), event.Repo, event.PullRequest, ghClientFunc)
+}
+
 func (h *autoMerger) handlePullRequestEvent(w http.ResponseWriter, event *github.PullRequestEvent, ghClientFunc GitHubIntegrationsClientFunc) error {
 	if strings.ToLower(event.GetAction()) != labeledEvent {
```

Comment:
Just to be sure that I got it right:

* In the `mergePR` function there is no check a label value 'approved' but if every of the required status is set to 'success'.
* So setting a label manually differently than "approved" just don't succeed if the status of the review has been already set to "approved".
* Setting a label manually to "approved" ... hmm .. that doesn't really do anything extra like approving the review. 

This means that the use case above with directly labelling with "approved' is not really supported.

So this check is only to trigger a potential merge. Think I got it :) (sorry for thinking loud)